source/BreachNode.cpp

```diff
@@ -15,18 +15,37 @@ constexpr float BREACH_SCALE = 3.0f;
 
 void BreachNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch, const Mat4& transform,
 					  Color4 tint) {
+	Vec2 breachPos;
 	if (breachModel->getHealth() > 0) {
-		Vec2 breachPos =
-			Vec2(globals::RADIUS * sin(breachModel->getAngle() * globals::PI_180),
-				 -(float)globals::RADIUS * cos(breachModel->getAngle() * globals::PI_180));
-		if (breachModel->getAngle() < 0) {
+		// Breach is currently active
+		float onScreenAngle = breachModel->getAngle() - playerDonutModel->getAngle();
+		onScreenAngle = onScreenAngle >= 0 ? onScreenAngle : shipSize + onScreenAngle;
+		onScreenAngle = onScreenAngle > shipSize / 2 ? onScreenAngle - shipSize : onScreenAngle;
+		onScreenAngle *= globals::PI_180;
+		if (!isShown && onScreenAngle < globals::SEG_CUTOFF_ANGLE &&
+			onScreenAngle > -globals::SEG_CUTOFF_ANGLE) {
+			// Breach is coming into visible range
+			float relativeAngle = onScreenAngle - getParent()->getParent()->getAngle();
+			breachPos = Vec2(globals::RADIUS * sin(relativeAngle),
+							 -(float)globals::RADIUS * cos(relativeAngle));
+			setPosition(breachPos);
+			setAngle(relativeAngle);
+			isShown = true;
+			CULog("Coming into view at %f", onScreenAngle / globals::PI_180);
```

Comment:
We'll probably want to take logs that occur with high frequency, such as ones from rolling around normally.

Reply:
oh sorry I meant to take those out..